File: HeartForCharity/HeartForCharity.Model/Requests/CampaignUpdateRequest.cs
using HeartForCharity.Model.Enums;
using System;

namespace HeartForCharity.Model.Requests
{
    public class CampaignUpdateRequest
    {
        public int? CategoryId { get; set; }
        public string Title { get; set; } = null!;
        public string? Description { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public decimal TargetAmount { get; set; }
        public CampaignStatus Status { get; set; }
    }
}

File: HeartForCharity/UI/heartforcharity_desktop/lib/model/requests/campaign_update_request.dart
class CampaignUpdateRequest {
  final int? categoryId;
  final String title;
  final String? description;
  final DateTime? startDate;
  final DateTime? endDate;
  final double targetAmount;
  final String? status;

  CampaignUpdateRequest({
    this.categoryId,
    required this.title,
    this.description,
    this.startDate,
    this.endDate,
    required this.targetAmount,
    this.status,
  });

  Map<String, dynamic> toJson() => {
        'categoryId': categoryId,
        'title': title,
        'description': description,
        'startDate': startDate?.toIso8601String(),
        'endDate': endDate?.toIso8601String(),
        'targetAmount': targetAmount,
        'status': status,
      };
}
